src/crowdin_client.cpp

```diff
@@ -57,56 +65,71 @@
 namespace
 {
 
-#define OAUTH_CLIENT_ID     "poedit-y1aBMmDbm3s164dtJ4Ur150e2"
-#define OAUTH_AUTHORIZE_URL "/oauth2/authorize?response_type=token&client_id=" OAUTH_CLIENT_ID
+#define OAUTH_SCOPE         "project+tm"
+#define OAUTH_CLIENT_ID     "k0uFz5HYQh0VzWgZmOpA"
+#define OAUTH_CLIENT_SECRET "p6Ko83cd4l4SD2TYeY8Fheffw8VuN3bML5jx7BeQ"
+//      any arbitrary unique unguessable string (e.g. UUID in hex)
+#define OAUTH_STATE         "948cf13ffffb47119d6cfa2b68898f67"
+//      The value of below macro should be set exactly as is (without quotes)
+//      to "Authorization Callback URL" of Crowdin application
+//      https://support.crowdin.com/enterprise/creating-oauth-app/
 #define OAUTH_URI_PREFIX    "poedit://auth/crowdin/"
-
-// Recursive extract files from /api/project/*/info response
-void ExtractFilesFromInfo(std::vector<std::wstring>& out, const json& r, const std::wstring& prefix)
-{
-    for (auto i : r["files"])
-    {
-        std::wstring name = prefix + str::to_wstring(i["name"]);
-        std::string node_type = i["node_type"];
-        if (node_type == "file")
-        {
-            out.push_back(name);
-        }
-        else if (node_type == "directory" || node_type == "branch")
-        {
-            ExtractFilesFromInfo(out, i, name + L"/");
-        }
-    }
-}
+#define OAUTH_AUTHORIZE_URL "/oauth/authorize" \
+	"?" "response_type" "=" "code" \
+	"&" "scope"         "=" OAUTH_SCOPE \
+	"&" "client_id"     "=" OAUTH_CLIENT_ID \
+	"&" "state"         "=" OAUTH_STATE \
+        "&" "redirect_uri"  "=" OAUTH_URI_PREFIX
 
 } // anonymous namespace
 
 
 std::string CrowdinClient::WrapLink(const std::string& page)
 {
-    std::string url("https://poedit.net/crowdin");
+    std::string url("https://accounts.crowdin.com");
     if (!page.empty() && page != "/")
-        url += "?u=" + http_client::url_encode(page);
+        url += page;
     return url;
 }
```

Comment:
This seems to repurpose `WrapLink`. It is meant to route Crowdin-heading links through poedit.net.

If the issue here is (I suspect) that poedit.net isn't correctly handling full URLs, please say so and I'll fix it.

Reply:
Addressed in #633 2166b6102400a216d5853e5c0b78a621cdfc1e63 seems exactly as we discussed in this PR plus something by email. I.e. source tracking params is added to Crowdin OAuth URL used without redirects from poedit.net